Jenkinsfile

```diff
@@ -0,0 +1,58 @@
+@Library('xmos_jenkins_shared_library@v0.27.0') _
+
+pipeline {
+    agent none
+
+    options {
+        disableConcurrentBuilds()
+        skipDefaultCheckout()
+        timestamps()
+        buildDiscarder(xmosDiscardBuildSettings(onlyArtifacts=false))
+    }
+    parameters {
+        string(
+            name: 'TOOLS_VERSION',
+            defaultValue: '15.2.1',
+            description: 'The XTC tools version'
+        )
+    }
+
+    stages {
+        stage("Do Jenkins") {
+            agent {
+                label 'linux&&64'
+            }
+            stages {
+                stage ("Do Jenkins") {
```

Comment:
You don't need so many nested stages->stage->stages->stage. You can set the agent on line 4 and then just take the stage from here onwards.

Reply:
done